libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -0,0 +1,295 @@
+        gait_pos_z[moving_leg] = 0
+        gait_pos_y[moving_leg] = gait_pos_z[moving_leg] - (y_speed/gait_speed_divisor)
+        gait_rot_z[moving_leg] = gait_rot_z[moving_leg] - (yaw_speed/gait_speed_divisor)
+    end
+end
+
+-- Body Forward Kinematics calculates where each leg should be.
+-- inputs are
+--   a) body rotations: body_rot_x, body_rot_y, body_rot_z
+--   b) body position: body_pos_x, body_pos_y, body_pos_z
+--   c) offset of the center of body
+function body_forward_kinematics(X, Y, Z, Xdist, Ydist, Zrot)
+    local totaldist = {X + Xdist + body_pos_x, Y + Ydist + body_pos_y}
+    local distBodyCenterFeet = math.sqrt(totaldist[1]^2 + totaldist[2]^2)
+    local AngleBodyCenter = math.atan(totaldist[2], totaldist[1])
+    local rolly = math.tan(math.rad(body_rot_y)) * totaldist[1]
+    local pitchy = math.tan(math.rad(body_rot_x)) * totaldist[2]
+
+    local ansx = math.cos(AngleBodyCenter + math.rad(body_rot_z+Zrot)) * distBodyCenterFeet - totaldist[1] + body_pos_x
+    local ansy = math.sin(AngleBodyCenter + math.rad(body_rot_z+Zrot)) * distBodyCenterFeet - totaldist[2] + body_pos_y
+    local ansz = rolly + pitchy + body_pos_z
+    return {ansx, ansy, ansz}
+end
+
+-- Leg Inverse Kinematics calculates the angles for each servo of each joint using the output of the
+-- body_forward_kinematics() function which gives the origin of each leg on the body frame
+function leg_inverse_kinematics(x, y, z)
+    local coxa = math.deg(math.atan(x, y))
+    local trueX = math.sqrt(x^2 + y^2) - COXA_LEN
+    local im = math.sqrt(trueX^2 + z^2)
+
+    local q1 = -math.atan(z, trueX)
+    local d1 = FEMUR_LEN^2 - TIBIA_LEN^2 + im^2
+    local d2 = 2*FEMUR_LEN*im
+    local q2 = math.acos(d1/d2)
+    local femur = math.deg(q1+q2)
+
+    d1 = FEMUR_LEN^2 - im^2 + TIBIA_LEN^2
+    d2 = 2*TIBIA_LEN*FEMUR_LEN
+    local tibia = math.deg(math.acos(d1/d2)-math.rad(90))
+    return {coxa, -femur, -tibia}
+end
+
+-- checks if the servo has moved to its expected postion 
+function servo_estimate(start_time,current,last_angle)
+    local target = 0
+    for j = 1, 12 do
+        curr_target = math.abs(current[j] - last_angle[j])
+        if curr_target > target then
+            target = curr_target
+        end
+    end
+    local target_time = target * (0.24/60) * 1000
+    return (target_time + start_time) <= millis()
+end
+
+-- main_inverse_kinematics produces the inverse kinematic solution for each
+-- leg joint servo by taking into consideration the initial_pos, gait offset and the body inverse kinematic values.
+function main_inverse_kinematics()
+    local ans1 = body_forward_kinematics(endpoints1[1]+gait_pos_x[1], endpoints1[2]+gait_pos_y[1], endpoints1[3]+gait_pos_z[1], FRAME_LEN/2, FRAME_WIDTH/2,gait_rot_z[1])
+    local angles1 = leg_inverse_kinematics(endpoints1[1]+ans1[1]+gait_pos_x[1],endpoints1[2]+ans1[2]+gait_pos_y[1], endpoints1[3]+ans1[3]+gait_pos_z[1])
+    angles1 = {-45 + angles1[1],angles1[2],angles1[3]}
```

Comment:
This is replacing the entire angles table with a new angles table, why not edit just the first index? (Again skipping making the second table). If you use the multiple returns suggestion above though then this is more obvious that you are just tweaking a single axis upwards.
```suggestion
    angles1[1] = -45 + angles1[1]
```

Reply:
the suggested fix has been included in the last commit, thanks again!